main.cpp

```diff
@@ -18,13 +18,27 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
-public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    Rewriter& rewrite;
+ public:
+    CastCallBack(Rewriter& rewriter) rewrite(rewriter) {};
 
     void run(const MatchFinder::MatchResult &Result) override {

```

Comment:
- How to prevent parentheses from being added if they are already present in the (int)(f) code?
- How to prevent changing the C style cast code for a special case - when we want to suppress warnings about unused variables?
- How to prevent changing the C style cast code in macros?

Reply:
1. We can do something like this:
```
auto rangeVariable = Lexer::getSourceText(CharSourceRange::getTokenRange(
                styleCastExpr->getRParenLoc().getLocWithOffset(1),
		styleCastExpr->getEndLoc()),
		*Result.SourceManager,
		LangOptions()).str();
	    auto originalSize = rangeVariable.size();
	    rangeVariable.erase(
	        remove_if(
	            rangeVariable.begin(),
	            rangeVariable.end(),
	            [](char symbol) { return symbol == ' '; }),
	       rangeVariable.end());
	       
	    if (rangeVariable.front() != '(' && rangeVariable.back() != ')') {
	    	rewriter.ReplaceText(rParenLoc.getLocWithOffset(1), originalSize, "(" + rangeVariable + ")");
	    } else {
	    	rewriter.ReplaceText(rParenLoc.getLocWithOffset(1), originalSize, rangeVariable);
	    }
```
2. We can do a simple check like this: `if (styleCastExpr->getCastKind() == CK_ToVoid)` and then do the cast or not depending on the result of this check.
3. By adding the corresponding check.

Reply:
`3. By adding the corresponding check.`
Code example is required.

Reply:
The required check can be as follows:
`if (CastExpression->getExprLoc().isMacroID()) return;`